tests/test_singlenodeexecutor_cache.py

```
@@ -34,6 +34,21 @@ def test_cache_data(self):
 
+    def test_cache_key(self):
+        cache_directory = "./cache"
+        with SingleNodeExecutor(cache_directory=cache_directory) as exe:
+            self.assertTrue(exe)
+            future_lst = [exe.submit(sum, [i, i], resource_dict={"cache_key": "same_" + str(i)}) for i in range(1, 4)]
+            result_lst = [f.result() for f in future_lst]
+
+        cache_lst = get_cache_data(cache_directory=cache_directory)
+        for entry in cache_lst:
+            self.assertTrue("same" in entry['filename'])
+        self.assertEqual(sum([c["output"] for c in cache_lst]), sum(result_lst))
```

Pylint (3.3.7)

[refactor] Consider using a generator instead 'sum(c['output'] for c in cache_lst)'

(R1728)